tests/app/authentication/test_jwt.py

```diff
@@ -0,0 +1,285 @@
+        with self.assertRaises(InvalidTokenException) as ite:
+            decoder.decode_signed_jwt_token(token_without_header)
+        self.assertIn("Missing Headers", ite.exception.value)
+
+    def test_jwt_does_not_contain_payload(self):
+        token_without_payload = jwtio_header + ".e30." + jwtio_signature
+        decoder = Decoder(PUBLIC_KEY, SR_PRIVATE_PEM, "digitaleq")
+        with self.assertRaises(InvalidTokenException) as ite:
+            decoder.decode_signed_jwt_token(token_without_payload)
+        self.assertIn("Signature verification failed", ite.exception.value)
+
+    def test_jwt_does_not_contain_signature(self):
+        jwt = jwtio_header + "." + jwtio_payload + ".e30"
+
+        decoder = Decoder(PUBLIC_KEY, SR_PRIVATE_PEM, "digitaleq")
+        with self.assertRaises(InvalidTokenException) as ite:
+            decoder.decode_signed_jwt_token(jwt)
+        self.assertIn("Signature verification failed", ite.exception.value)
+
+    def test_jose_header_missing_type(self):
+        header = base64.urlsafe_b64encode(b'{"alg":"RS256", "kid":"EDCRRM"}')
+        jwt = header.decode() + "." + jwtio_payload + "." + jwtio_signature
+
+        decoder = Decoder(PUBLIC_KEY, SR_PRIVATE_PEM, "digitaleq")
+        with self.assertRaises(InvalidTokenException) as ite:
+            decoder.decode_signed_jwt_token(jwt)
+        self.assertIn("Missing Type", ite.exception.value)
+
+    def test_jose_header_invalid_type(self):
+        header = base64.urlsafe_b64encode(b'{"alg":"RS256", "kid":"EDCRRM", "typ":"TEST"}')
+        jwt = header.decode() + "." + jwtio_payload + "." + jwtio_signature
+
+        decoder = Decoder(PUBLIC_KEY, SR_PRIVATE_PEM, "digitaleq")
+        with self.assertRaises(InvalidTokenException) as ite:
+            decoder.decode_signed_jwt_token(jwt)
+        self.assertIn("Invalid Type", ite.exception.value)
+
+    def test_jose_header_contains_multiple_type(self):
+        header = base64.urlsafe_b64encode(b'{"alg":"RS256", "kid":"EDCRRM","typ":"JWT","typ":"TEST"}')
+        jwt = header.decode() + "." + jwtio_payload + "." + jwtio_signature
+
+        decoder = Decoder(PUBLIC_KEY, SR_PRIVATE_PEM, "digitaleq")
+        with self.assertRaises(InvalidTokenException) as ite:
+            decoder.decode_signed_jwt_token(jwt)
+        self.assertIn("Multiple Type Headers", ite.exception.value)
+
+    def test_jose_header_missing_alg(self):
+        header = base64.urlsafe_b64encode(b'{"kid":"EDCRRM","typ":"JWT"}')
+        jwt = header.decode() + "." + jwtio_payload + "." + jwtio_signature
+
+        decoder = Decoder(PUBLIC_KEY, SR_PRIVATE_PEM, "digitaleq")
+        with self.assertRaises(InvalidTokenException) as ite:
+            decoder.decode_signed_jwt_token(jwt)
+        self.assertIn("Missing Algorithm", ite.exception.value)
+
+    def test_jose_header_invalid_alg(self):
+        header = base64.urlsafe_b64encode(b'{"alg":"invalid","kid":"EDCRRM","typ":"JWT"}')
```

Comment:
do we have a test for 'alg':'None', just because this was a known issue with JWT in the past